### ISSUE TITLE
Document API section of Reference

### ISSUE THREAD

**Issue by annasong20** [kubernetes-sigs/kustomize#5279]
This is a sub-task of #4338.

Fill in the [API section](https://github.com/kubernetes-sigs/kustomize/blob/master/site/content/en/docs/Reference/kustomization_file.md) under Reference. This will involve migrating [existing documentation of Kustomization fields](https://kubectl.docs.kubernetes.io/references/kustomize/kustomization/), as documented in [this spreadsheet](https://docs.google.com/spreadsheets/d/1Aaal0Ou10xeDkh_COWkg_r00R0I-Il_t5G2ad6W3jWc/edit#gid=0) and filling in any other gaps you see. Note also the [Notes](https://docs.google.com/document/d/1galldpW27fSyKr9JAdKcCHVJ2qEa40Mng1jTop7xong/edit?pli=1#heading=h.jh4o5c89dw5) on the Docs project.

Note the other Reference task #5278. Note also the structural changes to the Reference section in #5247. This is only my attempt to mimic the [Kubernetes doc](https://kubernetes.io/docs/reference/kubectl/) and is by no means authoritative. You can use the Kubernetes site as reference or choose a cleaner layout of preference. Solely as a nice to have, I'd be curious to see how we could automate this.

**Comment by bugoverdose** [comment]
I guess I should start working on this after #5247 is merged.
